Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5145,6 +5146,8 @@ void WebPageProxy::receivedNavigationActionPolicyDecision(WebProcessProxy& proce
         continueWithProcessForNavigation = WTFMove(continueWithProcessForNavigation)
     ](FrameProcess* sharedProcess) mutable {
         if (sharedProcess) {
+            navigation->setPendingSharedProcess(*sharedProcess);
```

Comment:
It seems like we should also call `setPendingSharedProcess(nullptr)` at some point right? It seems odd that a client that holds on to a `WKNavigationAction` object can cause a process to never die.